packages/salesforcedx-vscode-apex/src/index.ts

```diff
@@ -72,56 +72,64 @@ export async function activate(context: vscode.ExtensionContext) {
     coreTelemetryService.isTelemetryEnabled()
   );
 
-  const langClientHRStart = process.hrtime();
-  languageClient = await languageServer.createLanguageServer(context);
-  LanguageClientUtils.setClientInstance(languageClient);
-  const handle = languageClient.start();
-  context.subscriptions.push(handle);
+  // Initialize Apex language server
+  try {
+    const langClientHRStart = process.hrtime();
+    languageClient = await languageServer.createLanguageServer(context);
+    languageClientUtils.setClientInstance(languageClient);
+    const handle = languageClient.start();
+    languageClientUtils.setStatus(ClientStatus.Indexing, '');
```

Comment:
LanguageClientUtils is now a singleton that has an Unavailable status as its default. Once we set the language client instance and start it we set the status to indexing.